manta-crypto/src/eclair/bool.rs

```diff
@@ -96,6 +98,47 @@ where
 {
     /// Selects `true_value` when `bit == true` and `false_value` when `bit == false`.
     fn select(bit: &Bool<COM>, true_value: &Self, false_value: &Self, compiler: &mut COM) -> Self;
+
+    /// Selects an element from `table` by repeated iteration of `select` over `bits`.
+    /// The `bits` are ordered from most significant to least significant, forming unsigned integers in binary representation
+    /// which are understood as the `table` indices.
+    #[inline]
+    fn select_from_table<'s, B, T>(bits: B, table: T, compiler: &mut COM) -> Self
+    where
+        Self: 's + Clone,
+        Bool<COM>: 's,
+        B: IntoIterator<Item = &'s Bool<COM>>,
+        B::IntoIter: ExactSizeIterator,
+        T: IntoIterator<Item = &'s Self>,
+        T::IntoIter: ExactSizeIterator,
+    {
+        let mut table = table.into_iter();
+        let mut bits = bits.into_iter();
+        assert_eq!(
+            table.len(),
+            2usize.pow(bits.len() as u32),
```

Comment:
Does the system complain about just having `2.pow(bits.len())` because otherwise just switch to that.

Reply:
It doesn't compile, requires a u32.